torch/csrc/distributed/c10d/TCPStore.cpp

```diff
@@ -299,11 +303,39 @@ TCPStore::TCPStore(std::string host, const TCPStoreOptions& opts)
     // server successfully started
     C10D_DEBUG("The server has started on port = {}.", server_->port());
 
+    std::ifstream maxconnFile("/proc/sys/net/core/somaxconn");
+    if (maxconnFile.good() and numWorkers_.has_value()) {
+      try {
+        std::string str(
+            (std::istreambuf_iterator<char>(maxconnFile)),
+            std::istreambuf_iterator<char>());
+        std::size_t somaxconn = std::stoll(str);
+        if (somaxconn < *numWorkers_) {
+          C10D_WARNING(
+              "Starting store with {} workers but somaxconn is {}."
+              "This might cause instability during bootstrap, consider increasing it.",
+              *numWorkers_,
+              somaxconn);
+        }
+      } catch (std::exception& e) {
```

Comment:
its tempting to catch all here to avoid adding a new source of error to the application.  but should we? can you just check the types of exceptions thrown by stoll and only catch those?

Reply:
good point, it throws a bunch of exceptions, they all nail down to logic_error.